retriever/try_install_modified.py

```diff
@@ -0,0 +1,139 @@
+modified_scripts = get_modified_scripts()
+
+if modified_scripts is None:
+    print("No new scripts found. Database is up to date.")
+    sys.exit()
+if len(sys.argv) > 1:
+    ENGINE_LIST = [
+        e for e in ENGINE_LIST
+        if e.name in sys.argv[1:] or
+        e.abbreviation in sys.argv[1:]
+    ]
+
+if os.path.exists("test_modified"):
+    os.system("rm -r test_modified")
+os.makedirs("test_modified")
+os.chdir("test_modified")
+
+dbfile = os.path.normpath(os.path.join(os.getcwd(), 'testdb.sqlite'))
+
+engine_test = {
+    "postgres": {'engine': 'postgres',
+                 'user': 'postgres',
+                 'password': os_password,
+                 'host': 'localhost',
+                 'port': 5432,
+                 'database': 'postgres',
+                 'database_name': 'testschema',
+                 'table_name': '{db}.{table}'},
+
+    "mysql": {'engine': 'mysql',
+              'user': 'travis',
+              'password': '',
+              'host': 'localhost',
+              'port': 3306,
+              'database_name': 'testdb',
+              'table_name': '{db}.{table}'},
+
+    "xml": {'engine': 'xml',
+            'table_name': 'output_file_{table}.xml'},
+
+    "json": {'engine': 'json',
+             'table_name': 'output_file_{table}.json'},
+
+    "csv": {'engine': 'csv',
+            'table_name': 'output_file_{table}.csv'},
+
+    "sqlite": {'engine': 'sqlite',
+               'file': dbfile, 'table_name': '{db}_{table}'}
+}
+
+SCRIPT_LIST = SCRIPT_LIST()
+TEST_ENGINES = {}
+IGNORE = [
+    "forest-inventory-analysis",
+    "bioclim",
+    "prism-climate",
+    "vertnet",
+    "NPN",
+    "mammal-super-tree"
+]
+IGNORE = [dataset.lower() for dataset in IGNORE]
+
+for engine in ENGINE_LIST:
+    if engine.abbreviation in engine_test:
+        try:
+            opts = engine_test[engine.abbreviation]
+            TEST_ENGINES[engine.abbreviation] = choose_engine(opts)
+        except:
+            TEST_ENGINES[engine.abbreviation] = None
+            pass
+
+errors = []
+for module in MODULE_LIST:
+    for (key, value) in list(TEST_ENGINES.items()):
+        shortname = module.SCRIPT.shortname.lower()
+        if shortname in modified_scripts and shortname not in IGNORE:
+            if value is not None:
+                print("==>", module.__name__, value.name, "..........", module.SCRIPT.shortname)
+                try:
+                    module.SCRIPT.download(value)
+                except KeyboardInterrupt:
+                    pass
+                except Exception as e:
+                    print("ERROR.")
+                    errors.append((key, module.__name__, e))
+            else:
+                errors.append((key, "No connection detected......" + module.SCRIPT.shortname))
+
+if errors:
+    print("Engine, Dataset, Error")
+    for error in errors:
+        print(error)
+else:
+    print("All tests passed. All scripts are updated to latest version.")
+    update_version_file()
```

Comment:
@zhangcandrew, line 139 is calling the function which should update the script version numbers in ```version.txt```.